clients/src/main/java/org/apache/kafka/clients/producer/internals/ErrorLoggingCallback.java

```diff
@@ -45,7 +45,7 @@ public void onCompletion(RecordMetadata metadata, Exception e) {
             String valueString = (valueLength == -1) ? "null" :
                     logAsString ? new String(value) : valueLength + " bytes";
             log.error("Error when sending message to topic {} with key: {}, value: {} with error: {}",
-                    topic, keyString, valueString, e.getMessage());
+                    topic, keyString, valueString, e);
```

Comment:
I think you also need to remove the trailing `{}`. Exceptions get logged without requiring anything in the formatter string if they are the last trailing argument.